src/simulators.jl

```diff
@@ -137,13 +137,17 @@ function simulate!(sys,
     sim.remove_CM_motion && remove_CM_motion!(sys)
 
     for step_n in 1:n_steps
-        sys.coords += sys.velocities .* sim.dt .+ (remove_molar.(accels_t) .* sim.dt ^ 2) ./ 2
-        sys.coords = wrap_coords.(sys.coords, (sys.boundary,))
+        sys_copy = sys
```

Comment:
This doesn't actually copy the system? The two systems in the `run_constraints!(sys, sys_copy.coords, sim.dt)` line refer to the same thing. I would `copy(sys.coords)` if required.